src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerContext.cs

```diff
@@ -37,8 +37,11 @@ internal bool CanUseSerializationLogic
                             // Guard against unsupported features
                             Options.Converters.Count == 0 &&
                             Options.Encoder == null &&
-                            Options.NumberHandling == JsonNumberHandling.Strict &&
+                            (Options.NumberHandling & (JsonNumberHandling.WriteAsString | JsonNumberHandling.AllowNamedFloatingPointLiterals)) == 0 &&
```

Comment:
Why don't we support `NumberHandling == JsonNumberHandling.Strict` anymore?

Reply:
Oh, I missed the `== 0` part at the end. Maybe it would be good to add a comment here what the policy is?

Reply:
Yup, added a comment.